src/Shared/Traits.cs

```diff
@@ -27,6 +27,9 @@ public static Traits Instance
         public Traits()
         {
             EscapeHatches = new EscapeHatches();
+
+            DebugScheduler = DebugEngine || Environment.GetEnvironmentVariable("MSBUILDDEBUGSCHEDULER") != null;
```

Comment:
Why are these separate from all the readonly variables below?

Reply:
Language limitation:
![image](https://user-images.githubusercontent.com/2255729/125500387-e442530f-cce6-44c3-93e5-fa9d3718bf5f.png)